mathesar_ui/src/stores/table-data/records.ts

```diff
@@ -715,124 +740,43 @@ export class RecordsData {
-    }
-    this.updatePromises.set(cellKey, promise);
-
-    try {
-      const result = await promise;
-      this.meta.cellModificationStatus.set(cellKey, { state: 'success' });
-
-      const response: RpcResponse<RecordsResponse> = {
-        status: 'ok',
-        value: result,
-      };
-      this.updateSummaryStores([response]);
-
-      const updatedRecord = result.results[0];
-      const mergedRecord = this.mergePreviousJoinedColumnValues(
-        row.record,
-        updatedRecord,
-      );
+      .send();
 
-      return row.withRecord(mergedRecord);
-    } catch (err) {
-      this.meta.cellModificationStatus.set(cellKey, {
-        state: 'failure',
-        errors: [RpcError.fromAnything(err)],
-      });
-    } finally {
-      if (this.updatePromises.get(cellKey) === promise) {
-        this.updatePromises.delete(cellKey);
-      }
-    }
-    return row;
-  }
+    // We're silently swallowing errors. This might not be great for the user.
+    // We could consider ways of displaying this, but it might require some
+    // thought and design.
+    if (response.status === 'error') return;
+    const result = response.value;
 
-  async refetchAndMutateRow(
-    row: PersistedRecordRow,
-  ): Promise<PersistedRecordRow> {
-    const { record } = row;
-    const pkColumn = get(this.columnsDataStore.pkColumn);
-    if (pkColumn === undefined) {
-      // eslint-disable-next-line no-console
-      console.error('Unable to fetch record without a primary key column');
-      return row;
-    }
-    const primaryKeyValue = record[pkColumn.id];
-    if (primaryKeyValue === undefined) {
-      // eslint-disable-next-line no-console
-      console.error('Unable to fetch record with a missing primary key value');
-      return row;
-    }
-    const { joining } = get(this.meta.recordsRequestParamsData);
+    this.updateSummaryStores([{ status: 'ok', value: result }]);
 
-    try {
-      const result = await api.records
-        .get({
-          ...this.apiContext,
-          record_id: primaryKeyValue,
-          return_record_summaries: this.loadIntrinsicRecordSummaries,
-          ...joining.recordsRequestParams(),
-        })
-        .run();
-
-      const response: RpcResponse<RecordsResponse> = {
-        status: 'ok',
-        value: result,
-      };
-      this.updateSummaryStores([response]);
-
-      const updatedRecord = result.results[0];
-      const mergedRecord = this.mergePreviousJoinedColumnValues(
-        row.record,
-        updatedRecord,
-      );
-
-      return row.mutateRecord(mergedRecord);
-    } catch (err) {
-      // When error in refetching, return row as-is
-      // eslint-disable-next-line no-console
-      console.error('Error refetching row', err);
-      return row;
-    }
+    const updatedRecord = result.results[0];
+    this.fetchedRecordRows.update((rows) =>
+      rows.map((r) =>
+        r.identifier === rowId ? r.withRecord(updatedRecord) : r,
+      ),
+    );
```

Comment:
This method should also use the same logic from the `bulkUpdate` to only update the cells which do not have a request in process.